src/get-environments.ts

```diff
@@ -45,21 +45,22 @@ export async function getEnvironmentsForBranch() {
   }
 
   if (environments.length > 0) {
-    const { currentBranch, repository } = getGitRepoAndBranch();
+    const { currentBranch, repository, isDefaultBranch } =
+      await getGitRepoAndBranch();
     if (!currentBranch || !repository) {
       extensionState.onFailedToGetBranch();
       return [];
     } else {
       extensionState.setCurrentBranch(currentBranch);
     }
-    environments = environments.filter(
-      (environment) =>
-        repositoriesEqual(
-          environment?.latestDeploymentLog?.blueprintRepository,
-          repository
-        ) &&
-        environment?.latestDeploymentLog?.blueprintRevision === currentBranch
-    );
+    environments = environments.filter((environment) => {
+      const envRepository =
+        environment?.latestDeploymentLog?.blueprintRepository;
+      const envBranch = environment?.latestDeploymentLog?.blueprintRevision;
+      return repositoriesEqual(envRepository, repository) && envBranch
+        ? envBranch === currentBranch
+        : isDefaultBranch;
```

Comment:
nit pick, I prefer simpler conditions
```suggestion
      if( !repositoriesEqual(envRepository, repository)) return false;
      if ( isDefaultBranch ) return true;
      if ( envBranch ) return envBranch === currentBranch;
```

Reply:
94f0d2cfdb0fb08e89febd09da6eddef341c5be4
48ec4e6889d9df80fc593c6c9219bf8ac2c1d214